packages/amplify-cli-core/src/cfnUtilities.ts

```diff
@@ -1,12 +1,26 @@
-import * as yaml from 'js-yaml';
 import { Template } from 'cloudform-types';
-import { JSONUtilities } from './jsonUtilities';
 import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
 import * as path from 'path';
+import { JSONUtilities } from './jsonUtilities';
+
+export async function readCFNTemplate(filePath: string): Promise<{ templateFormat: CFNTemplateFormat; cfnTemplate: Template }>;
+export async function readCFNTemplate(
+  filePath: string,
+  options: { throwIfNotExist: boolean },
+): Promise<{ templateFormat: CFNTemplateFormat; cfnTemplate: Template } | undefined>;
+
+export async function readCFNTemplate(filePath: string, options: { throwIfNotExist: boolean } = { throwIfNotExist: true }) {
+  const errorMessage = `No CloudFormation template found at ${filePath}`;
+  if (!fs.existsSync(filePath)) {
+    if (options.throwIfNotExist === false) {
+      return undefined;
+    }
+    throw new Error(errorMessage);
+  }
 
-export async function readCFNTemplate(filePath: string): Promise<{ templateFormat: CFNTemplateFormat; cfnTemplate: Template }> {
-  if (!fs.existsSync(filePath) || !fs.statSync(filePath).isFile) {
-    throw new Error(`No CloudFormation template found at ${filePath}`);
+  if (!fs.statSync(filePath).isFile) {
```

Comment:
if the `throwIfNotExist` flag is set, I think undefined should be returned in this case as well